packages/core/src/config/load.ts

```diff
@@ -120,25 +135,106 @@ export function getLintRawConfigWithMergedContentByPriority(
   };
 }
 
-async function resolveExtends(lintConfig: LintRawConfig): Promise<LintRawConfig> {
-  if (!lintConfig.extends || !lintConfig.extends.length) return lintConfig;
+export function resolveNestedPlugins(
+  configPath: string,
+  pluginConfigPath: string,
+  plugin: string | Plugin,
+): string | Plugin | undefined {
+  if (!plugin) {
+    return;
+  }
+
+  if (!isString(plugin)) {
+    return plugin;
+  }
+
+  if (isAbsoluteUrl(plugin as string)) {
+    throw new Error(
+      `Error configuration format not detected (${plugin}). Nested plugin should be path to local file`,
+    );
+  }
+  const pluginPath = path.resolve(path.dirname(pluginConfigPath), plugin as string);
+
+  return path.resolve(path.dirname(configPath), pluginPath);
+}
+
+async function resolveApis(
+  apis: Record<string, Api>,
+  configPath: string = '',
+  resolve?: RawResolveConfig,
+): Promise<Record<string, Api>> {
+  const apiKeys = Object.keys(apis);
+  if (isEmptyArray(apiKeys)) {
+    return apis;
+  }
+
+  const resolvedApis: Record<string, Api> = {};
+  for (const apiKey of apiKeys) {
+    if (!apis[apiKey]?.lint?.extends) {
+      resolvedApis[apiKey] = apis[apiKey];
+    }
+    const lint = await resolveExtends(apis[apiKey].lint as LintRawConfig, configPath, resolve);
+    resolvedApis[apiKey] = {...apis[apiKey], lint};
+  }
+
+  return resolvedApis;
+}
+
+async function resolveExtends(
+  lintConfig: LintRawConfig | Omit<LintRawConfig, "plugins">,
+  configPath: string = '',
+  resolve?: RawResolveConfig,
+): Promise<LintRawConfig> {
```

Comment:
Seems like a better option:
```suggestion
async function resolveExtends({
  lintConfig: LintRawConfig | Omit<LintRawConfig, "plugins">,
  configPath: string = '',
  resolve?: RawResolveConfig,
}): Promise<LintRawConfig> {
```